interfaces/builtin/ppp_test.go

```diff
@@ -42,66 +40,87 @@ var _ = Suite(&PppInterfaceSuite{
 	iface: builtin.MustInterface("ppp"),
 })
 
-func (s *PppInterfaceSuite) SetUpTest(c *C) {
-	const mockPlugSnapInfo = `name: other
-version: 1.0
+const pppConsumerYaml = `name: consumer
 apps:
  app:
-  command: foo
   plugs: [ppp]
 `
 
-	const mockSlotSnapInfo = `name: ppp
-version: 1.0
-apps:
- app:
-  command: foo
-  slots: [ppp]
+const pppCoreYaml = `name: core
+type: os
+slots:
+  ppp:
 `
-	slotSnap := snaptest.MockInfo(c, mockPlugSnapInfo, nil)
-	s.slot = &interfaces.Slot{SlotInfo: slotSnap.Slots["ppp"]}
 
-	plugSnap := snaptest.MockInfo(c, mockPlugSnapInfo, nil)
-	s.plug = &interfaces.Plug{PlugInfo: plugSnap.Plugs["ppp"]}
+func (s *PppInterfaceSuite) SetUpTest(c *C) {
+	s.plug = MockPlug(c, pppConsumerYaml, nil, "ppp")
+	s.slot = MockSlot(c, pppCoreYaml, nil, "ppp")
 }
 
 func (s *PppInterfaceSuite) TestName(c *C) {
 	c.Assert(s.iface.Name(), Equals, "ppp")
 }
 
-func (s *PppInterfaceSuite) TestUsedSecuritySystems(c *C) {
-	apparmorSpec := &apparmor.Specification{}
-	err := apparmorSpec.AddConnectedPlug(s.iface, s.plug, nil, s.slot, nil)
-	c.Assert(err, IsNil)
-	c.Assert(apparmorSpec.SecurityTags(), DeepEquals, []string{"snap.other.app"})
-	c.Assert(apparmorSpec.SnippetForTag("snap.other.app"), testutil.Contains, `/usr/sbin/pppd ix,`)
-
-	apparmorSpec = &apparmor.Specification{}
-	err = apparmorSpec.AddPermanentSlot(s.iface, s.slot)
-	c.Assert(err, IsNil)
-	c.Assert(apparmorSpec.SecurityTags(), HasLen, 0)
-
-	dbusSpec := &dbus.Specification{}
-	err = dbusSpec.AddConnectedPlug(s.iface, s.plug, nil, s.slot, nil)
-	c.Assert(err, IsNil)
-	c.Assert(dbusSpec.SecurityTags(), HasLen, 0)
-	dbusSpec = &dbus.Specification{}
-	err = dbusSpec.AddPermanentSlot(s.iface, s.slot)
-	c.Assert(err, IsNil)
-	c.Assert(dbusSpec.SecurityTags(), HasLen, 0)
-
-	udevSpec := &udev.Specification{}
-	c.Assert(udevSpec.AddPermanentSlot(s.iface, s.slot), IsNil)
-	c.Assert(udevSpec.Snippets(), HasLen, 0)
+func (s *PppInterfaceSuite) TestSanitizeSlot(c *C) {
+	c.Assert(s.slot.Sanitize(s.iface), IsNil)
+	slot := &interfaces.Slot{SlotInfo: &snap.SlotInfo{
+		Snap:      &snap.Info{SuggestedName: "some-snap"},
+		Name:      "ppp",
+		Interface: "ppp",
+	}}
+
+	c.Assert(slot.Sanitize(s.iface), ErrorMatches,
+		"ppp slots are reserved for the core snap")
+}
+
+func (s *PppInterfaceSuite) TestSanitizePlug(c *C) {
+	c.Assert(s.plug.Sanitize(s.iface), IsNil)
+}
+
+func (s *PppInterfaceSuite) TestAppArmorSpec(c *C) {
+	spec := &apparmor.Specification{}
+	c.Assert(spec.AddConnectedPlug(s.iface, s.plug, nil, s.slot, nil), IsNil)
```

Comment:
Are we missing the permanent slot snippet? I'll check this out and correct this, if necessary.